pandas-stubs/io/pickle.pyi

```diff
@@ -1,16 +1,24 @@
 from __future__ import annotations
 
-from typing import Literal
+from typing import Any
 
-from pandas._typing import FilePathOrBuffer
+from pandas._typing import (
+    CompressionOptions,
+    FilePath,
+    ReadPickleBuffer,
+    StorageOptions,
+    WriteBuffer,
+)
 
 def to_pickle(
-    obj,
-    filepath_or_buffer: FilePathOrBuffer,
-    compression: str | None = ...,
+    obj: Any,
```

Comment:
I think `object` might be better here than `Any`. I think the rule of thumb is use `object` if it can literally be any object and use `Any` when it is too complex to type.

Reply:
Looking at typeshed they use `Any` both ways for `dump` and `load`.

https://github.com/python/typeshed/blob/d4287a7f08305d95d24d3e0487941ec9bb35f4ae/stdlib/pickle.pyi#L121-L128

Reply:
Seems object can cause problems with type checking 

https://stackoverflow.com/a/39817126/2551705

although this is probably more to do with loading than saving.